ldm/invoke/CLI.py

```diff
@@ -521,121 +558,205 @@ def set_default_output_dir(opt:Args, completer:Completer):
     completer.set_default_dir(opt.outdir)
 
 
-def add_weights_to_config(model_path:str, gen, opt, completer):
-    print(f'>> Model import in process. Please enter the values needed to configure this model:')
-    print()
+def import_model(model_path:str, gen, opt, completer):
+    '''
+    model_path can be (1) a URL to a .ckpt file; (2) a local .ckpt file path; or
+    (3) a huggingface repository id
+    '''
+    model_name = None
+    
+    if model_path.startswith(('http:','https:','ftp:')) or os.path.exists(model_path):
+        model_name = import_ckpt_model(model_path, gen, opt, completer)
+    elif re.match('^[\w-]+/[\w-]+$',model_path):
+        model_name = import_diffuser_model(model_path, gen, opt, completer)
+    else:
+        print(f'** {model_path} is neither the path to a .ckpt file nor a diffusers repository id. Can\'t import.')
 
-    new_config = {}
-    new_config['weights'] = model_path
+    if not model_name:
+        return
+        
+    if not _verify_load(model_name, gen):
+        print(f'** model failed to load. Discarding configuration entry')
+        gen.model_manager.del_model(model_name)
+        return
+    
+    if input('Make this the default model? [n] ') in ('y','Y'):
+        gen.model_manager.set_default_model(model_name)
+
+    gen.model_manager.commit(opt.conf)
+    completer.update_models(gen.model_manager.list_models())
+    print(f'>> {model_name} successfully installed')
+
+def import_diffuser_model(path_or_repo:str, gen, opt, completer)->str:
+    manager = gen.model_manager
+    default_name = Path(path_or_url).stem
```

Comment:
the argument's name `path_or_repo` but `path_or_url` is used, causing `!import_model` for diffusers to fail

Reply:
fixed.